presto-thrift/src/test/java/io/prestosql/plugin/thrift/integration/TestThriftProjectionPushdown.java

```diff
@@ -0,0 +1,199 @@
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Predicates.equalTo;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.prestosql.plugin.thrift.integration.ThriftQueryRunner.driftServerPort;
+import static io.prestosql.plugin.thrift.integration.ThriftQueryRunner.startThriftServers;
+import static io.prestosql.spi.type.VarcharType.VARCHAR;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.expression;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.project;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.tableScanWithMatchers;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static java.util.stream.Collectors.joining;
+
+public class TestThriftProjectionPushdown
+        extends BaseRuleTest
+{
+    private static final String CATALOG = "test";
+    private static final String TINY_SCHEMA = "tiny";
+    private List<DriftServer> servers;
+
+    private static final ThriftTableHandle NATION_THRIFT_TABLE = new ThriftTableHandle(new SchemaTableName(TINY_SCHEMA, "nation"));
+
+    private static final TableHandle NATION_TABLE = new TableHandle(
+            new CatalogName(CATALOG),
+            NATION_THRIFT_TABLE,
+            ThriftTransactionHandle.INSTANCE,
+            Optional.empty());
+
+    private static final Session SESSION = testSessionBuilder()
+            .setCatalog(CATALOG)
+            .setSchema(TINY_SCHEMA)
+            .build();
+
+    @Override
+    protected Optional<LocalQueryRunner> createLocalQueryRunner()
+    {
+        try {
+            servers = startThriftServers(1, false);
+        }
+        catch (Throwable t) {
+            if (servers != null) {
+                try {
+                    for (DriftServer server : servers) {
+                        server.shutdown();
+                    }
+                }
+                catch (Throwable e) {
+                    t.addSuppressed(e);
+                }
+            }
+
+            throw t;
+        }
+
+        String addresses = servers.stream()
+                .map(server -> "localhost:" + driftServerPort(server))
+                .collect(joining(","));
+
+        Map<String, String> connectorProperties = ImmutableMap.<String, String>builder()
+                .put("presto.thrift.client.addresses", addresses)
+                .put("presto.thrift.client.connect-timeout", "30s")
+                .put("presto-thrift.lookup-requests-concurrency", "2")
+                .build();
+
+        LocalQueryRunner runner = LocalQueryRunner.create(SESSION);
+        runner.createCatalog(CATALOG, getOnlyElement(new ThriftPlugin().getConnectorFactories()), connectorProperties);
+
+        return Optional.of(runner);
+    }
+
+    @AfterClass
+    public void cleanup()
+    {
+        if (servers != null) {
+            for (DriftServer server : servers) {
+                server.shutdown();
```

Comment:
Can you please use `Closer` or `io.airlift.testing.Closeables`. If not then I guess you need to try-catch every shutdown to prevent resource leaking.